### PR TITLE
XML attributes, XML boolean, route53 example

### PR DESCRIPTION
**Pull request by ltd**
* Added ability to pass a map to be used as XML attributes, needed for route53 api payload
* Added XML generation for boolean type
* Added examples directory with route53/change_rr.ex as first file

### CODE REVIEW

File: lib/aws/xml.ex [hunk 1]
```diff
@@ -61,6 +61,10 @@ defmodule AWS.XML do
     ["<", k, ">", Float.to_charlist(v), "</", k, ">"]
   end
 
+  defp encode_xml_key_value({k, v}) when is_binary(k) and is_boolean(v) do
```
Comment (philss) [inline lib/aws/xml.ex:64]: Could you add a test for this behavior?

File: lib/aws/xml.ex [hunk 2]
```diff
@@ -75,6 +79,30 @@ defmodule AWS.XML do
     ]
   end
 
+  # allow passing a map to use as xml attributes
+  defp encode_xml_key_value({{k, attrmap}, v})
+       when is_binary(k) and is_map(attrmap) and is_map(v) do
```
Comment (philss) [inline lib/aws/xml.ex:84]: Also a test for this one would be cool.

File: lib/aws/xml.ex (hunk 1 above)
Comment (ltd) [reply lib/aws/xml.ex:64]: will do

File: lib/aws/xml.ex (hunk 2 above)
Comment (ltd) [reply lib/aws/xml.ex:84]: will do

**Comment by philss** [comment]
@ltd thank you! :purple_heart: 

And sorry for the delay. :grimacing:

**Pull request merged**